tracing-attributes/tests/instrument.rs

```diff
@@ -80,8 +82,8 @@ fn fields() {
         .run_with_handle();
 
     with_default(subscriber, || {
-        my_fn(2, false);
-        my_fn(3, true);
+        my_fn(2, false, UnDebug(0), UnDebug(1));
```

Comment:
can we also add a `.only()` to the mock fields above, so that we assert no other fields are also added to the generated span? like:
```rust
            span.clone().with_field(
                field::mock("arg1")
                    .with_value(&format_args!("2"))
                    .and(field::mock("arg2").with_value(&format_args!("false")))
                    .only(),
            ),
```